unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BuildModeHUD/Scripts/BuildModeHUDController.cs

```diff
@@ -166,9 +176,68 @@ private void ConfigureCatalogItemDropController()
         catalogItemDropController.OnCatalogItemDropped += CatalogItemSelected;
     }
 
-    public void PublishStart() { view.PublishStart(); }
+    private void ConfigureBuildModeConfirmationModalController()
+    {
+        controllers.buildModeConfirmationModalController.OnCancelExit += CancelPublishModal;
+        controllers.buildModeConfirmationModalController.OnConfirmExit += ConfirmPublishModal;
+    }
+
+    public void PublishStart()
+    {
+        controllers.buildModeConfirmationModalController.Configure(
+            PUBLISH_MODAL_TITLE,
+            PUBLISH_MODAL_SUBTITLE,
+            PUBLISH_MODAL_CANCEL_BUTTON,
+            PUBLISH_MODAL_CONFIRM_BUTTON);
+        controllers.buildModeConfirmationModalController.SetActive(true, BuildModeModalType.PUBLISH);
+    }
+
+    internal void CancelPublishModal(BuildModeModalType modalType)
+    {
+        if (modalType != BuildModeModalType.PUBLISH)
+            return;
+
+        controllers.buildModeConfirmationModalController.SetActive(false, BuildModeModalType.PUBLISH);
+    }
+
+    internal void ConfirmPublishModal(BuildModeModalType modalType)
+    {
+        if (modalType != BuildModeModalType.PUBLISH)
+            return;
+
+        controllers.publishPopupController.PublishStart();
+        OnConfirmPublishAction?.Invoke();
+
+        // NOTE (Santi): This is temporal until we implement the way of return the publish progress from the kernel side.
+        //               Meanwhile we will display a fake progress.
+        publishProgressCoroutine = CoroutineStarter.Start(FakePublishProgress());
+    }
+
+    private IEnumerator FakePublishProgress()
+    {
+        while (true)
```

Comment:
We are 100% sure that `PublishEnd` is going to be called in the 100% of the cases? There's no possibility of any error getting the loading bar stuck forever here? I'd add some kind of bail out mechanism just in case.

Reply:
@BrianAmadori Once we press the [PUBLISH] button, unity side loss the control of the publication process and stay waiting until receive back the message `PublishSceneResult` from kernel (this message is who will raise the `PublishEnd()`, whether the publication has failed or not). Currently, the only place where could occur an error during the publish process is on the kernel side and if it happens it should propagate the error through the payload of the `PublishSceneResult` message. So I don't know what kind of bail out mechanism we could add from the unity side.

Reply:
@Suduck is there anything I can be missing here? Is this publication flow correct?

Reply:
As we talked yesterday during the Builder In World sync, I'm going to catch in the publish modal (Unity side) both kind of error messages: our generic static message as main title and the received one by kernel as subtitle.

Reply:
Done!